### PR TITLE
Fixes JSDoc for types generation

### PR DESCRIPTION
**Pull request by lemnis**
Helps fixing codeception/codeceptjs#2024 .

Intellisense will return 'any' for `CodeceptJS.LocatorOrString` within this repo as long the latest changes are not released yet for codeceptjs.

### CODE REVIEW

File: Detox.js [hunk 1]
```diff
@@ -395,8 +395,8 @@ class Detox extends Helper {
    * I.click({ ios: 'Save', android: 'SAVE' }, '#main'); // different texts on iOS and Android
    * ```
    * 
-   * @param {string|object} locator 
-   * @param {string|object} [context=null] 
+   * @param {CodeceptJS.LocatorOrString} locator 
+   * @param {?CodeceptJS.LocatorOrString} [context=null]
```
Comment (elukoyanov) [inline Detox.js]: What `?` means  in type?

File: Detox.js [hunk 2]
```diff
@@ -368,9 +368,9 @@ class Detox extends Helper {
    * I.longPress('Update', 2, '#menu'); // locate by text inside #menu, hold for 2 seconds
    * ```
    * 
-   * @param {string|object} locator element to locate
-   * @param {num} sec number of seconds to hold tap
-   * @param {string|object} context context element 
+   * @param {CodeceptJS.LocatorOrString} locator element to locate
+   * @param {number} sec number of seconds to hold tap
+   * @param {CodeceptJS.LocatorOrString} context context element
```
Comment (elukoyanov) [inline Detox.js]: context optional 
use `@param {CodeceptJS.LocatorOrString} [context]`

File: Detox.js [hunk 3]
```diff
@@ -349,9 +349,9 @@ class Detox extends Helper {
    * I.multiTap('Update', 2, '#menu'); // locate by id
    * ```
    * 
-   * @param {string|object} locator element to locate
-   * @param {int} num number of taps 
-   * @param {string|object} [context=null] context element
+   * @param {CodeceptJS.LocatorOrString} locator element to locate
+   * @param {number} num number of taps 
+   * @param {?CodeceptJS.LocatorOrString} [context=null] context element
```
Comment (elukoyanov) [inline Detox.js]: What `?` means in type? (2)

File: Detox.js [hunk 4]
```diff
@@ -505,8 +505,8 @@ class Detox extends Helper {
    * I.seeElementExists('~edit', '#menu'); // element inside #menu
    * ```
    * 
-   * @param {string|object} locator element to locate 
-   * @param {string|object} context  context element
+   * @param {CodeceptJS.LocatorOrString} locator element to locate 
+   * @param {CodeceptJS.LocatorOrString} context  context element
```
Comment (elukoyanov) [inline Detox.js]: opional use [context]

File: Detox.js [hunk 5]
```diff
@@ -487,8 +487,8 @@ class Detox extends Helper {
    * I.dontSeeElement('~edit'); // located by accessibility id
    * I.dontSeeElement('~edit', '#menu'); // element inside #menu
    * ```
-   * @param {string|object} locator element to locate
-   * @param {string|object} [context=null] context element
+   * @param {CodeceptJS.LocatorOrString} locator element to locate
+   * @param {?CodeceptJS.LocatorOrString} [context=null] context element
```
Comment (elukoyanov) [inline Detox.js]: What `?` means in type? (4)

File: Detox.js [hunk 6]
```diff
@@ -469,8 +469,8 @@ class Detox extends Helper {
    * I.seeElement('~edit', '#menu'); // element inside #menu
    * ```
    * 
-   * @param {string|object} locator element to locate 
-   * @param {string|object} [context=null] context element
+   * @param {CodeceptJS.LocatorOrString} locator element to locate 
+   * @param {?CodeceptJS.LocatorOrString} [context=null] context element
```
Comment (elukoyanov) [inline Detox.js]: What `?` means in type? (3)

File: Detox.js [hunk 7]
```diff
@@ -523,8 +523,8 @@ class Detox extends Helper {
    * I.dontSeeElementExist('~edit', '#menu'); // element inside #menu
    * ```
    * 
-   * @param {string|object} locator element to locate 
-   * @param {string|object} context context element
+   * @param {CodeceptJS.LocatorOrString} locator element to locate 
+   * @param {CodeceptJS.LocatorOrString} context context element
```
Comment (elukoyanov) [inline Detox.js]: optional, use `[context]`

File: Detox.js (hunk 1 above)
Comment (lemnis) [reply Detox.js]: `null` or

So basically it is the same as `null | CodeceptJS.LocatorOrString`

Comment (elukoyanov) [reply Detox.js]: nice, so, what if add `?` to all functions from this module, which allows `null`?

Comment (elukoyanov) [reply Detox.js]: longPress, seeElementExists, dontSeeElementExists

Comment (elukoyanov) [reply Detox.js]: I just don't understand, why generated docs does not include `| null` in types
image

**Comment by elukoyanov** [comment]
@lemnis, there are some merge conflicts, can you resolve them?

**Comment by lemnis** [comment]
Merge conflicts are solved know & changed the `?` to `| null` as it easier to understand

File: Detox.js [hunk 8]
```diff
@@ -505,8 +505,8 @@ class Detox extends Helper {
    * I.seeElementExists('~edit', '#menu'); // element inside #menu
    * ```
    * 
-   * @param {string|object} locator element to locate 
-   * @param {string|object} context  context element
+   * @param {CodeceptJS.LocatorOrString} locator element to locate 
+   * @param {CodeceptJS.LocatorOrString} [context=null]  context element
```
Comment (elukoyanov) [inline Detox.js]: One  `|  null ` is forgotten.
But we may add it later

**Comment by elukoyanov** [comment]
@DavertMik , ready for merge and release

**Comment by DavertMik** [comment]
Let's get it merged! :+1:

**Pull request merged**